### PR TITLE
Issue 86 customer realm scopes

### LINKED ISSUE #86: CustomerUserRealm: grant all scopes allowed by the client
We need to grant automatically all requested scopes when using the CustomerUserRealm (calling customer web service) if they are allowed by the client.

Example: client A uses implicit flow to get a token for customer X with scope "read-customer-profile", this scope can be used by resource servers for authorization. The scope "read-customer-profile" has no value here and is just used as a boolean flag.

### PR DESCRIPTION
**Pull request by hjacobs**
Fixes #86:

* make sure that customer realm can request any scope allowed by client
* make sure that customer realm cannot request more scopes than allowed by client (important for implicit grant flow where no client authentication takes place)

### CODE REVIEW

**Comment by hjacobs** [comment]
@danieldelhoyo @vroldanbet feel free to review :smile:

File: src/main/java/org/zalando/planb/provider/realms/ClientRealm.java [hunk 1]
```diff
@@ -5,8 +5,21 @@
 import java.util.Optional;
 import java.util.Set;
 
+import static java.util.stream.Collectors.toSet;
+
 public interface ClientRealm extends Realm {
```
Comment (vroldanbet) [inline src/main/java/org/zalando/planb/provider/realms/ClientRealm.java]: nice :)

**Comment by vroldanbet** [comment]
looks good to me! +1

**Comment by vroldanbet** [comment]
+1

**Comment by vroldanbet** [comment]
1

**Comment by vroldanbet** [comment]
+1

**Comment by vroldanbet** [comment]
:+1:

**Comment by vroldanbet** [comment]
:+1:

**Comment by vroldanbet** [comment]
:+1:

**Pull request merged**